### DISCUSSION TITLE
Save intermediate images before the final result.

### DISCUSSION THREAD

**Discussion by narcibrook** [lllyasviel/Fooocus#2878] [Q&A] [answered]
Hello, there are some images during the generation process that I really like, is there a temporary folder where I can find them?
thank you for your response ^^

**Comment by xtiondiv123** [comment]
hi，there is a folder which named output. It contained all the pictures and log .
![1715041148290(1)](https://github.com/lllyasviel/Fooocus/assets/152345887/367218c0-9d2a-4b0b-9b6a-867e35402287)

**Reply by narcibrook** [reply]
Ok thank you for your answer, but I expressed myself badly, these are the intermediate noise images that I want to recover without having to stay in front and download them one by one during generation ^^
![image](https://github.com/lllyasviel/Fooocus/assets/109109007/7f2c1c20-b511-4d2c-bd23-d1e0117276a5)

**Reply by mashb1t** [reply, marked as answer]
@narcibrook these intermediate images are not rendered in full resolution and use an approximative VAE. What you can do is to memorize the step which you liked and generate the same image again until this exact step with the same seed applied.
This is the best way i can think of you can get these images.

**Comment by mavcook** [comment]
I also wanted to save these, as some steps produce more interesting results than the final image. Also, this gives you the ability to make an animated sequence of the generation progress.

Add the code below after this line for a quick hack https://github.com/lllyasviel/Fooocus/blob/5a71495822a11bbabf7c889eed6d9b38b261bb96/modules/core.py#L300
```
            from PIL import Image
            img = Image.fromarray(y)
            img.save(f'/tmp/booya_step-{step}.png')
```
I don't know if this is the best place to put the image saving, or a decent way to do it, but it works. Obviously you can make this a lot better. I have something locally that outputs the intermediate images along in the same output folder with the same naming convention as the final image, but it required some more refactoring.
